engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/IfThenElseNode.java

```diff
@@ -0,0 +1,154 @@
+package org.enso.interpreter.node.controlflow;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Cached.Shared;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.Node.Child;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.InlinedCountingConditionProfile;
+import java.util.Objects;
+import org.enso.interpreter.node.ExpressionNode;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.EnsoObject;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.error.PanicSentinel;
+import org.enso.interpreter.runtime.warning.AppendWarningNode;
+import org.enso.interpreter.runtime.warning.WarningsLibrary;
+
+@NodeInfo(
+    shortName = "if_then_else",
+    description = "The runtime representation of a if then else expression.")
+public final class IfThenElseNode extends ExpressionNode {
+  private @Child ExpressionNode cond;
+  private @Child ExpressionNode trueBranch;
+  private @Child ExpressionNode falseBranch;
+  private @Child WithCondition with;
+
+  IfThenElseNode(ExpressionNode cond, ExpressionNode trueBranch, ExpressionNode falseBranch) {
+    this.cond = Objects.requireNonNull(cond);
+    this.trueBranch = trueBranch;
```

Comment:
Add `Objects.requireNonNull(trueBranch)` as well? I understand that only `falseBranch` can be null.

Reply:
Done in 933d74ca7e